### PR TITLE
Papers: add published papers to repository

### PR DESCRIPTION
**Pull request by michaelpj**
- Adds the published papers to our repostiory.
- Removes some unnecessary files.
- Updates everything to work with Agda 2.6.0

I want to get this in ASAP so I can update the paper to point to this repository for the artifacts.

### CODE REVIEW

**Comment by kwxm** [comment]
Looks fine to me.  We could just add the PDFs and any artefacts, but I come from the era when storage space was at a premium.

In fact, I see that these just contain the source.  Should we add PDFs to simplify things for people who just want to read things without having to build them?

Also, i see that the submission guidelines (https://www.cs.nott.ac.uk/~pszgmh/mpc19.html#submission) say

"The proceedings of MPC 2019 will be published in the Lecture Notes in Computer Science (LNCS) series, as with all previous instances of the conference. Authors of accepted papers will be expected to transfer copyright to Springer for this purpose."

Are we sure that it's OK to publish the final version here?  

I also have a vague feeling that I once saw something about IOHK wanting to retain ownership of publications, but I may be imagining that.  I'll see if I can find anything about that.

File: papers/unraveling-recursion/compiling-to-fomega.tex [hunk 1]
```diff
@@ -0,0 +1,733 @@
...
+\newtoggle{lagda}
+\togglefalse{lagda}
+% This is a bit of a hack to make it easy to toggle on including
+% the generated literate Agda latex if it's been built:
+% - Run agda --latex over all the files
+% - echo '\toggletrue{lagda}' > agdaswitch.tex
+% - Built the latex
+\InputIfFileExists{agdaswitch}{}{}
+
+\iftoggle{lagda}{
```
Comment (kwxm) [inline papers/unraveling-recursion/compiling-to-fomega.tex]: I don't remember the details, but wasn't this required to overcome some deficiency in an old version of Agda?  Can we do away with this now?  I think there was a problem with unexpected bits of Agda appearing in the PDF if you used the wrong value of the toggle.

Comment (kwxm) [reply papers/unraveling-recursion/compiling-to-fomega.tex]: Nah, maybe that's irrelevant.  I think that the problem before was that `\begin{code}[hide]` didn't work with the version of Agda style file that we were using. Perhaps it does work now.  That won't remove the need for the `\iftoggle` though, it'll just make it work properly.

**Comment by jmchapman** [comment]
I think there is a distinction between the final submitted version and the publisher's own version. The authors often have greater freedom with the former. I cannot find the LNCS rules though.

Thank you for doing this.

Will this pull request allow me to give a stable link to the latex source (a.k.a. the literate agda program)?

File: papers/unraveling-recursion/compiling-to-fomega.tex (hunk 1 above)
Comment (michaelpj) [reply papers/unraveling-recursion/compiling-to-fomega.tex]: No, it's there so you can have fast turnaround during development by not having Agda in the loop.

**Comment by michaelpj** [comment]
> We could just add the PDFs and any artefacts, but I come from the era when storage space was at a premium.

Checking in the PDFs probably takes more space. Besides, I don't believe in checking in generated files ;)

> Should we add PDFs to simplify things for people who just want to read things without having to build them?

We can add links to hydra.

I agree with James that we should be fine with publishing them, they're just like preprints.

> Will this pull request allow me to give a stable link to the latex source (a.k.a. the literate agda program)?

Yes, that's what I'm planning to do. You can take the github link at a particular commit, which should be stable and also reflect the published version.

File: papers/unraveling-recursion/compiling-to-fomega.tex (hunk 1 above)
Comment (kwxm) [reply papers/unraveling-recursion/compiling-to-fomega.tex]: Yes, I remembered that later.  There was definitely something broken though (see my second comment), and that may well have gone away now.

**Comment by michaelpj** [comment]
Okay, I'll do pinning later, but I want to get the submission updated asap.

**Pull request merged**